### PR TITLE
Bubble sort homework

### PR DESCRIPTION
**Pull request by nigel-riss**

### CODE REVIEW

**Comment by nigel-riss** [comment]
https://bubble-sort-nigel-riss.netlify.app/

File: 02-basic-typescript/bubble/src/utils/gen.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  maxLength: number = Options.MAX_LENGTH,
+  minValue: number = Options.MIN_VALUE,
+  maxValue: number = Options.MAX_VALUE,
+): Array<BubbleItem> => {
+  const arrLength = getRandomInt(minLength, maxLength);
+  const arr: Array<BubbleItem> = [...new Array(arrLength)]
+    .map((_, i: number): BubbleItem => ({
+      id: i,
+      value: getRandomInt(minValue, maxValue),
+    }));
```
Comment (shpakkdv) [inline 02-basic-typescript/bubble/src/utils/gen.ts]: Just a suggestion to simplify a bit:
```ts
Array.from({ length: arrLength }).map(() => {});
```
or even:
```ts
Array.from({ length: arrLength }, () => {});
```

File: 02-basic-typescript/bubble/src/components/Bar.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  value,
+  maxValue,
+}: BarProps): React.ReactElement => {
+  return(<div
+    className="bar"
+    style={{
+      height: `${value / maxValue * 100}%`
+    }}
+  >
+    {/* {value} */}
```
Comment (shpakkdv) [inline 02-basic-typescript/bubble/src/components/Bar.tsx]: Is it needed?

File: 02-basic-typescript/bubble/src/components/App.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  constructor(props: {}) {
+    super(props);
+
+    this.state = {
+      isSolved: false,
+      isPaused: true,
+      iteration: 0,
+      solution: [[]],
+    }
+  }
```
Comment (shpakkdv) [inline 02-basic-typescript/bubble/src/components/App.tsx]: Also, can be written without `constructor`:
```ts
    state = {
      isSolved: false,
      isPaused: true,
      iteration: 0,
      solution: [[]],
    }
```

**Review by shpakkdv — APPROVED** [review]

**Comment by shpakkdv** [comment]
The implementation looks really cool! Awesome idea!

File: 02-basic-typescript/bubble/src/utils/gen.ts (hunk 1 above)
Comment (nigel-riss) [reply 02-basic-typescript/bubble/src/utils/gen.ts]: Thx! Nice one.

File: 02-basic-typescript/bubble/src/components/Bar.tsx (hunk 2 above)
Comment (nigel-riss) [reply 02-basic-typescript/bubble/src/components/Bar.tsx]: I tried to display bar value. But it doesn't look nice with large arrays. I'll clean up it.

**Pull request merged**